thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/api/detection/AnomalyDetectionResource.java

```diff
@@ -0,0 +1,756 @@
+    metricConfigDTO.setOnlineData(this.objectMapper.writeValueAsString(dataNode));
+
+    this.metricConfigValidator.validateConfig(metricConfigDTO);
+
+    metricConfigDAO.save(metricConfigDTO);
+    LOG.info("Created metric with config {}", metricConfigDTO);
+
+    return metricConfigDTO;
+  }
+
+  DetectionConfigDTO generateDetectionConfig(JsonNode payloadNode, String suffix,
+      DatasetConfigDTO datasetConfigDTO, MetricConfigDTO metricConfigDTO, long start, long end) {
+    DetectionConfigDTO detectionConfigDTO;
+    Map<String, Object> detectionYaml;
+    ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+
+    if (payloadNode.has(DETECTION_FIELD)) {
+      // Customized configuration: retrieve config from user request
+      detectionYaml = ConfigUtils.getMap(yaml.load(payloadNode.get(DETECTION_FIELD).textValue()));
+    } else {
+      // Default configuration: retrieve the template from disk
+      detectionYaml =
+          ConfigUtils.getMap(yaml.load(classLoader.getResourceAsStream(TEMPLATE_DETECTION_PATH)));
+    }
+
+    // Do not support customized detection name as it is not a common use case
+    detectionYaml.put(DETECTION_YAML_FIELD, DEFAULT_DETECTION_NAME + suffix);
+    detectionYaml.put(DATASET_YAML_FIELD, datasetConfigDTO.getName());
+    detectionYaml.put(METRIC_YAML_FIELD, metricConfigDTO.getName());
+
+    detectionConfigDTO =
+        new DetectionConfigTranslator(this.yaml.dump(detectionYaml), this.provider).translate();
+    detectionConfigDTO.setCron("0 0 0 1 1 ? 2200"); // Never scheduled
+
+    // Tune the detection config - Passes the raw yaml params & injects tuned params
+    DetectionConfigTuner detectionTuner = new DetectionConfigTuner(detectionConfigDTO, provider);
+    detectionConfigDTO = detectionTuner.tune(start, end);
+
+    // Validate the detection config
+    detectionValidator.validateConfig(detectionConfigDTO);
+
+    detectionConfigDAO.save(detectionConfigDTO);
+    LOG.info("Created detection with config {}", detectionConfigDTO);
+
+    return detectionConfigDTO;
+  }
+
+  TaskDTO generateTaskConfig(long detectionConfigId, long start, long end)
+      throws JsonProcessingException {
+    TaskDTO taskDTO = new TaskDTO();
+    taskDTO.setJobName(TaskConstants.TaskType.DETECTION.toString() + "_" + detectionConfigId);
+    taskDTO.setStatus(TaskConstants.TaskStatus.WAITING);
+    taskDTO.setTaskType(TaskConstants.TaskType.DETECTION_ONLINE);
+    DetectionPipelineTaskInfo taskInfo =
+        new DetectionPipelineTaskInfo(detectionConfigId, start, end);
+    String taskInfoJson = objectMapper.writeValueAsString(taskInfo);
+    taskDTO.setTaskInfo(taskInfoJson);
+
+    taskDAO.save(taskDTO);
+    LOG.info("Created task: {}", taskDTO);
+
+    return taskDTO;
+  }
+
+  private TaskDTO pollingTask(long taskId) {
+    long startTime = System.currentTimeMillis();
+    TaskDTO taskDTO;
+
+    // Timeout mechanism will be handled by worker thread in the controller
```

Comment:
I'll still put a timeout mechanism here so that if anything external goes wrong, the thread can still timeout.